### PR TITLE
SNOW-931952: Provide IncludeRetryReason config param

### PR DESCRIPTION
**Pull request by sfc-gh-pfus**
### Description
Provided IncludeRetryReason param to disable sending http status code with retries query requests.

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: dsn.go [hunk 1]
```diff
@@ -99,6 +99,8 @@ type Config struct {
 	ClientStoreTemporaryCredential ConfigBool // When true the ID token is cached in the credential manager. True by default in Windows/OSX. False for Linux.
 
 	DisableQueryContextCache bool // Should HTAP query context cache be disabled
+
+	IncludeRetryReason bool // Should retried request contain retry reason
```
Comment (sfc-gh-dprzybysz) [inline dsn.go:103]: should we add here information about the default value?

File: dsn_test.go [hunk 2]
```diff
@@ -608,6 +609,20 @@ func TestParseDSN(t *testing.T) {
 			ocspMode: ocspModeFailOpen,
 			err:      nil,
 		},
+		{
+			dsn: "u:p@a.r.c.snowflakecomputing.com/db/s?account=a.r.c&includeRetryReason=true",
```
Comment (sfc-gh-dprzybysz) [inline dsn_test.go:613]: let's add a test with explicit false for the new parameter

File: retry_test.go [hunk 3]
```diff
@@ -227,7 +227,56 @@ func TestRetryQuerySuccess(t *testing.T) {
...
 		client,
-		emptyRequest, urlPtr, make(map[string]string), 60*time.Second, constTimeProvider(123456)).doPost().setBody([]byte{0}).execute()
+		emptyRequest, urlPtr, make(map[string]string), 60*time.Second, constTimeProvider(123456), nil).doPost().setBody([]byte{0}).execute()
+	if err != nil {
+		t.Fatal("failed to run retry")
+	}
+	var values url.Values
+	values, err = url.ParseQuery(urlPtr.RawQuery)
+	if err != nil {
+		t.Fatal("failed to fail to parse the URL")
```
Comment (sfc-gh-dprzybysz) [inline retry_test.go:237]: failed to fail? is it a good message?

Comment (sfc-gh-pfus) [reply retry_test.go]: Fixed.

File: dsn_test.go (hunk 2 above)
Comment (sfc-gh-pfus) [reply dsn_test.go:649]: Done.

File: dsn.go (hunk 1 above)
Comment (sfc-gh-dprzybysz) [reply dsn.go]: but should we document it somewhere?

Comment (sfc-gh-pfus) [reply dsn.go]: I believe that no. This is actually more like a hidden feature, for emergency use cases. I don't think that users should switch it on their own.

**Review by sfc-gh-dprzybysz — APPROVED** [review]

**Review by sfc-gh-dheyman-1 — APPROVED** [review]

**Pull request merged**